pkg/agent/handler/accessrequest.go

```diff
@@ -206,6 +219,27 @@ func (h *accessRequestHandler) onPending(ctx context.Context, ar *management.Acc
 	return ar
 }
 
+func (h *accessRequestHandler) provision(par *provAccReq) (prov.RequestStatus, prov.AccessData) {
+	status, accessData := h.prov.AccessRequestProvision(par)
+	if status.GetStatus() == prov.Success {
+		return status, accessData
+	}
+
+	for i := range h.retryCount {
+		// 7(sleep) + 3(time to finish AppProvision)*4(nb of calls) = 19s, should not exceed the watch-controller timeout(30s)
```

Comment:
Is watch controller waiting for an ack? Why the timeout? 

Perhaps we should Ack that we got the event, and then work to handle it, as our handling does not go back over that connection but through new calls to API Server

Reply:
I ask because I think our retries should be spaced larger than the seconds listed.